W-16124485: Addding clarification about custom configuration properties providers not being capable of using ConnectionProviders, hence not supporting connectivity testing. (#2808)
mulesoft/docs-mule-runtime-3: modules/ROOT/pages/custom-configuration-properties-provider.adoc

== Using Custom Configuration Properties Provider versus a Connector

<<<<<<< W-16124485
For static properties, use the properties provider approach, because static properties do not change during runtime. If your properties might change during runtime, create a connector that can provide the value as one of its operations.

[NOTE]
Although the xref:mule-sdk::getting-started.adoc[Mule SDK for Java] is used to implement a custom configuration properties provider, you can't use xref:mule-sdk::connections.adoc[ConnectionProviders] to connect to external sources. This is because `ConnectionProviders` are managed by Mule runtime and are initialized at a later stage, after the static properties are resolved. Consequently, connectivity testing isn't supported for custom properties providers.
=======
For static properties, use the properties provider approach, because static properties don't change during runtime. If your properties might change during runtime, create a connector that can provide the value as one of its operations.

> >>>>>>> v4.7

Reviewer: ```suggestion
For static properties, use the properties provider approach, because static properties don't change during runtime. If your properties might change during runtime, create a connector that can provide the value as one of its operations.

[NOTE]
Although the xref:mule-sdk::getting-started.adoc[Mule SDK for Java] is used to implement a custom configuration properties provider, you can't use xref:mule-sdk::connections.adoc[ConnectionProviders] to connect to external sources. This is because `ConnectionProviders` are managed by Mule runtime and are initialized at a later stage, after the static properties are resolved. Consequently, connectivity testing isn't supported for custom properties providers.
```

== Using Custom Configuration Properties Provider versus a Connector

For static properties, use the properties provider approach, because static properties do not change during runtime. If your properties might change during runtime, create a connector that can provide the value as one of its operations.

> NOTE: Even though the xref:mule-sdk::getting-started.adoc[Mule Java SDK] is used to implement a custom configuration properties provider, you can't use xref:mule-sdk::connections.adoc[ConnectionProviders] to connect to external sources. This is because `ConnectionProviders` are meant to be managed by the Mule Runtime, and they get initialized at a later stage than when the static properties need to be resolved. This also means connectivity testing is also not supported for custom properties providers.

Reviewer: ```suggestion
[NOTE]
Although the xref:mule-sdk::getting-started.adoc[Mule SDK for Java] is used to implement a custom configuration properties provider, you can't use xref:mule-sdk::connections.adoc[ConnectionProviders] to connect to external sources. This is because `ConnectionProviders` are managed by Mule runtime and are initialized at a later stage, after the static properties are resolved. Consequently, connectivity testing isn't supported for custom properties providers.
```
